File: backend/src/main/java/com/gpuoj/model/Submission.java
package com.gpuoj.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Table("submissions")
public class Submission {
    @Id
    private UUID id;
    private UUID userId;
    private String problemId;
    private String sourceCode;
    private String language = "CUDA";
    private String status = "PENDING";
    private String verdict;
    private Integer wallTimeMs;
    private Integer peakVramMb;
    private Double speedup;
    private String stdout;
    private String stderr;
    private OffsetDateTime submittedAt;
}

File: backend/src/main/kotlin/com/gpuoj/model/Submission.kt
package com.gpuoj.model

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import java.time.OffsetDateTime
import java.util.UUID

@Table("submissions")
data class Submission(
    @Id var id: UUID? = null,
    var userId: UUID? = null,
    var problemId: String? = null,
    var sourceCode: String? = null,
    var language: String = "CUDA",
    var status: String = "PENDING",
    var verdict: String? = null,
    var wallTimeMs: Int? = null,
    var peakVramMb: Int? = null,
    var speedup: Double? = null,
    var stdout: String? = null,
    var stderr: String? = null,
    var submittedAt: OffsetDateTime? = null
)
